packages/sample-app/package.json

```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@monorepo/sample-app",
+  "version": "0.0.0-fixed",
+  "description": "Sample plugin host application project",
+  "private": true,
+  "scripts": {
+    "clean": "rm -rf dist",
+    "build": "yarn clean && yarn webpack",
+    "build-prod": "yarn clean && NODE_ENV=production yarn webpack",
+    "lint": "yarn run -T eslint $INIT_CWD",
+    "webpack": "node -r ts-node/register ./node_modules/.bin/webpack",
+    "http-server": "http-server dist -p 9000 -c-1"
+  },
+  "devDependencies": {
+    "@openshift/dynamic-plugin-sdk": "portal:../lib-core",
+    "@openshift/dynamic-plugin-sdk-webpack": "portal:../lib-webpack",
```

Comment:
@karelhala

Replacing
```json
    "@openshift/dynamic-plugin-sdk": "portal:../lib-core",
    "@openshift/dynamic-plugin-sdk-webpack": "portal:../lib-webpack",
```
with
```json
    "@openshift/dynamic-plugin-sdk": "*",
    "@openshift/dynamic-plugin-sdk-webpack": "*",
```
will cause Yarn to pull these dependencies from npmjs public registry, resulting in `packages/sample-app/node_modules/@openshift` directory, which is something we don't want.

The `portal:` protocol is meant to refer to local packages, which is what we really need here.

The reason why `@openshift/dynamic-plugin-sdk` and `@openshift/dynamic-plugin-sdk-webpack` dependencies are present in the sample app's manifest:
- make it explicit that this project depends on plugin SDK packages (in real apps, you **_don't_** want to use `portal:` protocol, but a proper semver range)
- `@openshift/dynamic-plugin-sdk` is part of sample app provided shared modules (`packages/sample-app/src/plugin-sdk-addons/shared-modules.ts`) and webpack `ModuleFederationPlugin` will complain if it's missing in the manifest

Reply:
This serisouly smells like bad implementation on yarn side of things. If we were to use npm 7+ the workspace package hoisting works as it should. Meaning if you are referncing a package from within workspace it will autolink it on install [workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces). I am seriously advising to adopt npm soon-ish, maybe once this PR is merged we can focus on migrating away from yarn.